### ISSUE TITLE
Ability to use just total and update without passing an iterable

### ISSUE THREAD

**Issue by TheCleric** [cloud-oak/ProgressBars.jl#45]
In tqdm the following code is possible:

```python
from tqdm import tqdm

progress = tqdm(total=100)

for x in range(10):
    for y in range(10):
        progress.update()
```

This comes in handy for having one progress bar for nested for loops or for a progress bar that can be updated as threads return. It would be nice for ProgressBars to support this as well.

**Comment by khdlr** [comment]
Agreed! Imo we should also extend this to support `progress.update(Int)` for usecases like download progress, etc.

The use you suggested would then be a special case of that, where the argument of `progress.update` defaults to 1.

I'll see how much work this will actually take, shouldn't be too bad.

**Comment by TheCleric** [comment]
> The use you suggested would then be a special case of that, where the argument of `progress.update` defaults to 1.

Yes, exactly.
